connect/mirror/src/main/java/org/apache/kafka/connect/mirror/MirrorSourceConnector.java

```diff
@@ -581,13 +582,26 @@ void incrementalAlterConfigs(Map<String, Config> topicConfigs) {
         }));
     }
 
-    private void updateTopicAcls(List<AclBinding> bindings) {
-        log.trace("Syncing {} topic ACL bindings.", bindings.size());
-        targetAdminClient.createAcls(bindings).values().forEach((k, v) -> v.whenComplete((x, e) -> {
-            if (e != null) {
-                log.warn("Could not sync ACL of topic {}.", k.pattern().name(), e);
-            }
-        }));
+    // Visible for testing
+    int updateTopicAcls(List<AclBinding> bindings) {
+        Set<AclBinding> addBindings = new HashSet<>(bindings);
+        Set<AclBinding> failedBindings = new HashSet<>();
+        addBindings.removeAll(knownTopicAclBindings);
+        int newBindCount = addBindings.size();
+        if (!addBindings.isEmpty()) {
+            log.info("Syncing new found {} topic ACL bindings.", newBindCount);
+            targetAdminClient.createAcls(addBindings).values().forEach((k, v) -> v.whenComplete((x, e) -> {
+                if (e != null) {
+                    log.warn("Could not sync ACL of topic {}.", k.pattern().name(), e);
+                    failedBindings.add(k);
+                }
+            }));
+            knownTopicAclBindings = new HashSet<>(bindings);
+            knownTopicAclBindings.removeAll(failedBindings);
```

Comment:
Isn't `failedBindings` likely to be empty at this point unless the admin client is able to perform the request to create the ACL bindings exceptionally fast (which also happens to be the scenario we're covering in the unit test)?